react-example/src/components/Arkose.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class Arkose extends React.Component {
```

Comment:
can we please use a functional component instead of class based? The other components in the example are functional so this one is inconsistent.

Reply:
````
import React, { forwardRef, useEffect, useImperativeHandle } from 'react';
import PropTypes from 'prop-types';

const Arkose = forwardRef((props, ref) => {
  useImperativeHandle(ref, () => ({
    setupEnforcement() {
      setupEnforcement();
    },
  }));

  // Append the JS tag to the Document Body.
  const loadScript = () => {
    const scriptId = 'arkose-script';
    const currentScript = document.getElementById(scriptId);
    if (currentScript) {
      currentScript.remove();
    }
    const script = document.createElement('script');
    script.id = scriptId;
    script.type = 'text/javascript';
    script.src = `[https://client-api.arkoselabs.com/v2/${props.publicKey}/api.js](https://client-api.arkoselabs.com/v2/$%7Bprops.publicKey%7D/api.js)`;
    script.setAttribute('data-callback', 'setupEnforcement');
    script.async = true;
    script.defer = true;
    if (props.nonce) {
      script.setAttribute('data-nonce', props.nonce);
    }
    document.body.appendChild(script);
    return script;
  };

  const setupEnforcement = (myEnforcement) => {
    ref.current.myEnforcement = myEnforcement;
    ref.current.myEnforcement.setConfig({
      selector: props.selector && `#${props.selector}`,
      mode: props.mode,
      onReady: () => {
        console.log('READY');
        if (props.onReady) {
          props.onReady();
        }
      },
      onShown: () => {
        console.log('onShown');
        if (props.onShown) {
          props.onShown();
        }
      },
      onShow: () => {
        if (props.onShow) {
          props.onShow();
        }
      },
      onSuppress: () => {
        if (props.onSuppress) {
          props.onSuppress();
        }
      },
      onCompleted: (response) => {
        if (props.onCompleted) {
          props.onCompleted(response.token);
        }
      },
      onReset: () => {
        if (props.onReset) {
          props.onReset();
        }
      },
      onHide: () => {
        if (props.onHide) {
          props.onHide();
        }
      },
      onError: (response) => {
        if (props.onError) {
          props.onError(response?.error);
        }
      },
      onFailed: (response) => {
        if (props.onFailed) {
          props.onFailed(response);
        }
      },
    });
  };

  useEffect(() => {
    console.log('mount');
    // Anything in here is fired on component mount.

    const scriptElement = loadScript();
    // This will inject required html and css after the Arkose script is properly loaded
    scriptElement.onload = () => {
      console.log('Arkose API Script loaded');
      window.setupEnforcement = setupEnforcement.bind(this);
    };
    // If there is an error loading the Arkose script this callback will be called
    scriptElement.onerror = () => {
      console.log('Could not load the Arkose API Script!');
    };
    return () => {
      console.log('unmount');
      // Anything in here is fired on component unmount.
      if (window.setupEnforcement) {
        delete window.setupEnforcement;
      }
    };
  }, []);

  return <>{props.mode === 'inline' && <div id={props.selector}></div>}</>;
});

Arkose.propTypes = {
  publicKey: PropTypes.string.isRequired,
  mode: PropTypes.oneOf(['inline', 'lightbox']),
  selector: PropTypes.string,
  nonce: PropTypes.string,
  onReady: PropTypes.func,
  onShown: PropTypes.func,
  onShow: PropTypes.func,
  onSuppress: PropTypes.func,
  onCompleted: PropTypes.func,
  onReset: PropTypes.func,
  onHide: PropTypes.func,
  onError: PropTypes.func,
  onFailed: PropTypes.func,
};
export default Arkose;
```

Reply:
I discussed this with @hemangd we'll leave this as is for now, but will re-evaluate when we have more time